config/batch_inference.yaml

```diff
@@ -0,0 +1,14 @@
+ft_model_id: sayakpaul/gemma-2b-sft-qlora-no-robots
+ft_model_revision: main
```

Comment:
The problem with using branch name as a revision is that even if the branch is updated we won't know if it was updated. A better way could be to get the commit info from the repository and use the SHA as the revision since a commit SHA can be passed to the `revision` argument in `from_pretrained()`. 

But this is a just a good practice. Okay if we not do it. In that case, maybe going back to the previous state of this PR (no model revision) would be a better solution.

Reply:
automatically retrieved from what? branch name? I am confused

Reply:
Already mentioned:

> A better way could be to get the commit info from the repository and use the SHA as the revision since a commit SHA can be passed to the revision argument in from_pretrained().

Reply:
Getting the latest commit info and retrieve SHA automatically? 
I mean the revision(SHA or branch name) is a sort of user's choice. How to decide which commit info to grasp?

Reply:
Sorry for not being clear. 

Take a look at this line:
https://github.com/deep-diver/llmops-pipeline/blob/ed3db07df1723de6737ce4ead80e39018a27ac2f/src/gen/local_lm.py#L19

It accepts a model id to load from the Hub. In case it's not a local directory, we should query the repository to get its commit SHA and use it as a revision to store as model details. This is a better way to inform the stakeholders of a project about the exact state of the model that was used. In the case of a local directory to use for model loading, I think we can keep things as is as that way, it's almost impossible to know which state the model was in (it can be changed locally without knowing, too). 

We shouldn't let the user pass any `revision` because it's not resilient. I have already explained why in the comments above. 

LMK if there's still anything unclear.

Reply:
Sounds like it is not a model loading issue but recording the revision into the dataset. It would be OK to have branch name to download & load the model to perform batch inference, but at the same time the exact SHA should be recorded since ***branch name*** can not distinguish from commit to commit.

Do I understand correct?

Reply:
Yes.

Reply:
Thanks for your elaboration. Really appreciate it. 
In that case, I need to find out how to retrieve SHA in Python then :)

Reply:
Leverage the `huggingface_hub` API:

https://huggingface.co/docs/huggingface_hub/v0.10.0.rc3/en/package_reference/hf_api

Reply:
got it `model_info()`
: https://huggingface.co/docs/huggingface_hub/v0.10.0.rc3/en/package_reference/hf_api#huggingface_hub.HfApi.model_info